ui/src/data_explorer/containers/DataExplorer.js

```diff
@@ -79,15 +84,24 @@ const DataExplorer = React.createClass({
       queryConfigs,
       queryConfigActions,
       source,
+      hideWriteFormAction,
+      showWriteFormAction,
+      dataExplorer: {showWriteForm},
     } = this.props
     const {activeQueryIndex} = this.state
 
     return (
       <div className="data-explorer">
+        {showWriteForm
```

Comment:
I think I over-engineered this to avoid using component state, but this is a situation where component state is probably a better place for it.